File: 프로그래머스/3/43162. 네트워크/네트워크.py
def bfs(graph,start,visited):
    queue=[]
    queue.append(start)
    visited[start]=True
    
    while queue:
        curr=queue.pop()
        
        for neighbor in graph[curr]:
            if visited[neighbor]==False:
                queue.append(neighbor)
                visited[neighbor]=True
    



def solution(n, computers):
    graph=[[] for _ in range(n)]
    visited=[False]*n
    
    for i in range(n):
        for j in range(n):
            if i!=j and computers[i][j]==1:
                graph[i].append(j)
                graph[j].append(i)
                graph[i]=list(set(graph[i]))
                graph[j]=list(set(graph[j]))
                
    cnt=0
    for i in range(n):
        if visited[i]==False:
            bfs(graph,i,visited)
            cnt+=1
        
    return cnt

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    
    public static void dfs(int node,int[][] computers,boolean[] visited){

        visited[node]=true;
        
        for(int i=0;i<computers.length;i++){
            if(!visited[i] && computers[node][i]==1){
                dfs(i,computers,visited);
            }
        }
        
        
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        int cNum = computers.length; //컴퓨터 개수
        boolean[] visited = new boolean[cNum];
        
        for (int i=0;i<cNum;i++){
            if (!visited[i]){
                answer++;
                dfs(i,computers,visited);   
            }
        }
        return answer;
    }
}